### ISSUE TITLE
Improve code quality comments according to guidelines

### ISSUE THREAD

**Issue by stackoversnow** [OpenCovenant/quill#89]
As a novice contributor found it hard to extrapolate the purpose of functions. When making minor contributions to a small part of a small component, one shouldn't have to extrapolate the meaning of all other functions from reading the code only. This is easily alleviated by adding JSDoc based comments to the function, classes, properties etc,. The comment alone gives insight of what the method (or class , etc) does and inputs and outputs it produces. That gives enough knowledge to just know that that function execution or the specific class is not related to the issue being solved.
Also it avoids cases where one developer owns all the knowledge of previous code decisions which influence future changes, as this knowledge is put in the comments and shared instead.

As a line of work I'd suggest the following: 
1. Agree to follow a common comment guideline: the [Google Javascript Style guide](https://google.github.io/styleguide/jsguide.html#jsdoc) for comments and other guidelines
2. Add this guideline in the **README** file or where applicable as a notice to future contributors as a rule to maintain long term code style quality
3. Add PR label: **JSDOC** or *Code Comments* or some other name, to note all future PRs dealing with adding JSDoc to the codebase. Related PRs should only change/add comments and not touch any logic.

**Comment by AndiBraimllari** [comment]
Agreed that adding JSDocs in many places would improve things overall. I've recently created a branch `improve_code_docs` and will add there soon some changes regarding this. Also, many things can clearly be refactored, e.g. #93.

Regarding the style guide I guess any is fine with me, but perhaps we can decide this gradually, similarly for other changes.

We can keep this issue open and track it during development.
